### ISSUE TITLE
New issue after the fix for "muiName" error in @jbrowse/react-linear-genome-view

### ISSUE THREAD

**Issue by yeweiyang96** [GMOD/jbrowse-components#4614]
After the initial error related to muiName was fixed, a new issue has appeared when using @jbrowse/react-linear-genome-view. This issue also appears in the _jbrowse-react-linear-genome-view-vite-demo_ repository.
However, this problem does not occur when using **Webpack**.

<img width="1227" alt="截屏2024-10-23 14 48 51" src="https://github.com/user-attachments/assets/3d37a846-877c-4f67-8142-169a7c47ff21">

I got this error from JBrowse, here is the stack trace:

```
Error: Element type is invalid. Received a promise that resolves to: [object Object]. Lazy element type must resolve to a class or function.
../../react-dom/cjs/react-dom.development.js:20042:19 (mountLazyComponent@)
../../react-dom/cjs/react-dom.development.js:27465:24 (beginWork$1@)
../../react-dom/cjs/react-dom.development.js:26596:24 (performUnitOfWork@)
../../react-dom/cjs/react-dom.development.js:26505:23 (workLoopSync@)
../../react-dom/cjs/react-dom.development.js:26473:20 (renderRootSync@)
../../react-dom/cjs/react-dom.development.js:25889:35 (recoverFromConcurrentError@)
../../react-dom/cjs/react-dom.development.js:26135:47 (performSyncWorkOnRoot@)
../../react-dom/cjs/react-dom.development.js:12042:31 (flushSyncCallbacks@)
../../react-dom/cjs/react-dom.development.js:25690:32 (@)
```

**Comment by yeweiyang96** [comment]
Would you recommend temporarily switching to Webpack until this issue is resolved, or is there another known workaround for Vite?

**Comment by cmdcolin** [comment]
there was another thread that was made reporting this just after you. i fixed it and it is pending release in the next version. https://github.com/GMOD/jbrowse-react-linear-genome-view-vite-demo/issues/2 fixed by https://github.com/GMOD/jbrowse-components/pull/4612 there is no hotfix for this but i should release that fix in the next day or two

i generally like to have all our library work in any sort of bundler you try to throw at it (hence the matrix at https://jbrowse.org/jb2/docs/embedded_components/) but it is just a funny thing where vite is a tricky one to support. it comes up with some very funky ones. i even reported the bug from this thread here to the MUI team https://github.com/mui/material-ui/issues/44180 but i think ultimately it is like a esbuild "bug" perhaps or maybe a "limitation" of esbuild for using our CJS build (our modules are not yet "pure ESM" and from what i can tell esbuild is using our commonjs aka CJS build)

anyways, that's more info than you want to know. I think webpack is a good system, but we want to make vite a first class platform so we'll get a fix out for this ASAP

**Comment by cmdcolin** [comment]
(to be clear, esbuild is used by vite in dev mode, so that's why i mention it above)

**Comment by cmdcolin** [comment]
should be fixed in v2.16.0 :)
